app/src/main/java/org/mozilla/focus/widget/Cursor.java

```diff
@@ -67,6 +80,7 @@ public Point getLocation() {
 
     public void moveCursor(Direction direction) {
         activeDirections.add(direction);
+        cursorHandler.removeCallbacks(cursorTick);
```

Comment:
If we stick with Handlers, I think we might want to remove the callbacks in `onStop` too – otherwise:
- The Runnable gets added to the handler queue and could run while the fragment is in the background (waste of resources)
- The Runnable has a strong reference to the Context/Activity (being an anonymous class of the View), which is a large object, creating a memory leak if the cursor is recreated often (this can happen on "configuration change", when the device is rotated or keyboard added, etc., for which Android destroys the active activity and recreates it; note: I think we override this behavior for Focus so maybe this is less relevant). This is a common Android problem ([here's where it was first mentioned](https://android-developers.googleblog.com/2009/01/avoiding-memory-leaks.html) with static variables).
- Keeping a strong reference, the Runnable can call animate on a detached view hierarchy – I don't think this would crash but it's a waste of resources and could be confusing while debugging if there are multiple runnables.

You could fix the memory leak (which is incrementally better) by not using an anonymous class and creating a `WeakReference` to the view (we already do this for Activity and Context in various places in the code base). But this is probably more complex than just calling onStart/onStop.

But then again, hooking this up to the onStart/onStop callbacks is a pain in the ass and sort of breaks encapsulation, so maybe it's not worth the complexity. @liuche What do you think?